device/common/include/traccc/fitting/device/impl/fit.ipp

```diff
@@ -34,8 +34,7 @@ TRACCC_HOST_DEVICE inline void fit(
         return;
     }
 
-    const unsigned int param_id =
-        param_ids.at(static_cast<unsigned int>(globalIndex));
+    const unsigned int param_id = param_ids.at(globalIndex);
```

Comment:
This should be `global_index_t`.

Reply:
Should it? We take an element out of the `param_ids` vector. It has `unsigned int` values, true, but these are still different things from `global_index_t`. 🤔

Reply:
I guess it depends a bit on the semantics of the `global_index_t` type; if the semantics are exactly that they represent thread identifiers and nothing else then yes, although then I feel it might also be useful to have a separate index type for array indices. :thinking:

Reply:
We could start using `global_index_t` in a number of other places as well. But as long as the `param_ids` variable uses `unsigned int` directly, we should use `unsigned int` here as well. (We'd need to switch to using `global_index_t` for the `param_ids` variables everywhere before I'd want to change the type on this line.)